api/include/opentelemetry/logs/logger_type_traits.h

```diff
@@ -145,6 +146,10 @@ struct LogRecordSetterTrait<common::KeyValueIterable>
 template <class ValueType>
 struct LogRecordSetterTrait
 {
+  static_assert(!std::is_same<nostd::unique_ptr<LogRecord>, ValueType>::value &&
+                    !std::is_same<std::unique_ptr<LogRecord>, ValueType>::value,
+                "unique_ptr<LogRecord> is not allowed");
```

Comment:
```suggestion
                "unique_ptr<LogRecord> is not allowed, use std::move()");
```

Also, consider adding an assert for `LogRecord*`

Reply:
Thanks, it's done